backend/src/main/kotlin/cz/loono/backend/api/service/ExaminationRecordService.kt

```diff
@@ -0,0 +1,132 @@
+package cz.loono.backend.api.service
+
+import cz.loono.backend.api.dto.ExaminationTypeEnumDto
+import cz.loono.backend.data.model.Account
+import cz.loono.backend.data.model.ExaminationRecord
+import cz.loono.backend.data.repository.AccountRepository
+import cz.loono.backend.data.repository.ExaminationRecordRepository
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+
+@Service
+class ExaminationRecordService @Autowired constructor(
+    private val accountRepository: AccountRepository,
+    private val examinationRecordRepository: ExaminationRecordRepository,
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Loads all the examination records for this account `uid`.
+     *
+     * If a record for an examination type is not found, a default record for that type is inserted.
+     *
+     * If the database contains records with types that no longer exist, they will be removed from the database.
+     *
+     * Since missing examination records are inserted and leftover records are purged,
+     * this method always returns a record per each existing type.
+     */
+    @Transactional(rollbackFor = [Exception::class])
+    fun getOrCreateRecords(uid: String): List<ExaminationRecord> {
```

Comment:
If I understand the concept. You want to keep all types of examinations for each account. Am I right? Why you want to keep all these records ahead?

Reply:
I wanted to always return all examinations with up-to-date content, such that the client application can rely on this API to return valid data without any further processing on the client side.